ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEvents.scala

```diff
@@ -26,11 +28,11 @@ private[events] object EventsTableFlatEvents {
             contractId = contractId,
             templateId = templateId,
             createArgument = createArgument,
-            createSignatories = createSignatories,
-            createObservers = createObservers,
+            createSignatories = ArraySeq.unsafeWrapArray(createSignatories),
+            createObservers = ArraySeq.unsafeWrapArray(createObservers),
             createAgreementText = createAgreementText,
             createKeyValue = createKeyValue,
-            eventWitnesses = eventWitnesses,
+            eventWitnesses = ArraySeq.unsafeWrapArray(eventWitnesses),
```

Comment:
Does this situation call for unsafe operations, or should we use `.to(ArraySeq)` here instead?

Reply:
Afaict this is safe since we get the array directly from the parser and don’t let it escape anywhere. It looked like it might be performance critical for some cases of the ledger not do have a conversion here but I don’t know if this is actually a bottleneck or not. @gerolf-da what do you think?

In case it’s not obvious, we need this because for 2.13 scalapb generates code expecting an immutable Seq and `Array` is not implicitly converted to that. Alternatively we could also change what scalapb but immutable seqs seems very sensible to me.

Reply:
@mziolekda: Could I redirect this question to you please?

Reply:
This should not be a bottleneck, strictly speaking, but it would probably be good to be performance-conscious here.

Another alternative would be to use `anorm.SqlParser.list` instead of `anorm.SqlParser.array` to extract fields, but using `unsafeWrapArray` seems to be the most performance-conscious one, but without an experiment this is just an opinion. For now I would keep the behavior as close as possible to the existing one.

Reply:
Does `unsafeWrapArray` count as keeping it as close to the existing one as possible or are you suggesting something else? I’m not quite sure how or if the `Array` got converted before.

Reply:
looking through it, it seems like it did a copy before so I guess I could just throw `toIndexedSeq` in a few places which I believe happened implciitly before but I have a hard time seeing how that could possible be more efficient than not copying.

Reply:
Mmmh, I suggested to "the behavior as close as possible to the existing one" in the hope that the previous operation actually skipped the copy. If `unsafeWrapArray` works I would go for it. Perhaps adding a comment as to why it's ok to use it here would be useful for future readers.

Reply:
I think you might be right. For 2.13, I get
```
ledger/participant-integration-api/src/main/scala/platform/store/dao/events/Raw.scala:75: warning: method copyArrayToImmutableIndexedSeq in class LowPriorityImplicits2 is deprecated (since 2.13.0): Implicit conversions from Array to immutable.IndexedSeq are implemented by copying; Use the more efficient non-copying ArraySeq.unsafeWrapArray or an explicit toIndexedSeq call
        signatories = createSignatories,
```
So the 2.13 default is to copy which clearly sucks. The 2.12 default is to use `WrappedArray` since there we still operate on collection.Seq which afaict does not copy.

So `unsafeWrappedArray` seems as close as we can get provided we want to operate on an immutable Seq.

The other option would be not require an immutable Seq (by changing the scalapb config at least for the event module) but that seems kind of shitty given that 1. scalapb claims in the docs that they expect things to be immutable and 2. mutability is scary :).

Reply:
Added a comment on the safety of `unsafeWrapArray`.